File: 算法/Golang/（2）冒泡排序.go
package main
import "fmt"

func bubbleSort(data []int) []int {

	for i:=0;i<len(data);i++  {

		for j:=0;j<len(data)-i-1;j++  {
			//这里的j+1决定了为什么内层循环要-1
			if  data[j]>data[j+1]{
				temp := data[j+1]
				data[j+1] = data[j]
				data[j] = temp
			}
		}
	}
	return  data
}

func main()  {
	arr := []int{7,8,9,4,5,7,4,1,2,3,5,6}
	rel := bubbleSort(arr)
	fmt.Println(rel)
}

File: 算法/C++/（2）冒泡排序.cpp
#include <iostream>
#include <vector>

using namespace std;

vector<int> bubbleSort(vector<int> data) {

	int i, j, temp;

	for (i = 0; i < data.size(); i++) {

		for (j = 0; j < data.size() - i - 1; j++) {
			if (data[j] > data[j+1])
			{
				temp = data[j + 1];
				data[j + 1] = data[j];
				data[j] = temp;
			}
		}
	}
	return data;
}

int main() {

	vector<int> data;
	data = { 3,2,1 };
	data = bubbleSort(data);

	for (const int& k : data) {
		cout << k << endl;
	}

}